From 1cf8485850c39b1e4f86b473ff47565aaa5fc6c9 Mon Sep 17 00:00:00 2001
From: Sumit Jamadar <xkbr38@motorola.com>
Date: Mon, 9 Nov 2015 19:21:31 -0600
Subject: [PATCH 465/959] IKMMINTG-6609: arm/dt: 8952: Update camera power up
 sequence

P1 or later boards have new camera gpio/regulators
connections. So update camera power up sequence
for p1.

Restore P0 camera power up sequence delta in
p0 product dts file to keep camera functional.

Change-Id: I5383f0e4e54e0b6e2507b28026b9a79618777b56
Signed-off-by: Sumit Jamadar <xkbr38@motorola.com>
Reviewed-on: http://gerrit.mot.com/798516
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Dmitry Ornatsky <ornatsky@motorola.com>
Reviewed-by: Satya Prasad Kottisa <fvgm76@motorola.com>
Reviewed-by: Christopher Fries <cfries@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 arch/arm/boot/dts/qcom/msm8952-athene-p0.dts       | 16 +++++++++++++++
 .../boot/dts/qcom/msm8952-camera-sensor-moto.dtsi  | 23 +++++++++++++---------
 2 files changed, 30 insertions(+), 9 deletions(-)

diff --git a/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts b/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
index cab1f9c2620..0cd9ce3ab89 100644
--- a/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
+++ b/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
@@ -29,6 +29,22 @@
 };
 
 &cci {
+	qcom,camera@0 {
+		cam_vdig-supply = <&pm8950_l23>;
+		qcom,cam-vreg-min-voltage = <1200000 1800000>;
+		qcom,cam-vreg-max-voltage = <1200000 1800000>;
+		qcom,cam-vreg-op-mode = <1200000 1800000>;
+		/delete-property/ gpios;
+		gpios = <&msm_gpio 26 0>,
+			<&msm_gpio 36 0>;
+		/delete-property/ qcom,gpio-vdig;
+		qcom,gpio-req-tbl-num = <0 1>;
+		qcom,gpio-req-tbl-flags = <1 0>;
+		/delete-property/ qcom,gpio-req-tbl-label;
+		qcom,gpio-req-tbl-label = "CAMIF_MCLK0",
+					  "CAM_RESET0";
+	};
+
 	st,stmvl53l0
 	{
 		cell-index = <1>;
diff --git a/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi b/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
index 3308e3ba112..68673091e77 100644
--- a/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
@@ -43,23 +43,28 @@
 		qcom,mount-angle = <90>;
 		qcom,led-flash-src = <&led_flash0>;
 		qcom,actuator-src = <&actuator0>;
-		cam_vdig-supply = <&pm8950_l23>;
+		cam_vdig-supply = <&pm8950_s4>;
 		cam_vio-supply = <&pm8950_l10>;
 		qcom,cam-vreg-name = "cam_vdig", "cam_vio";
-		qcom,cam-vreg-min-voltage = <1200000 1800000> ;
-		qcom,cam-vreg-max-voltage = <1200000 1800000>;
-		qcom,cam-vreg-op-mode = <1200000 1800000>;
+		/* camera needs s4 regulator min 1.85v but audio
+		 * 8952_wcd_codec uses same regulator at min 2.05v */
+		qcom,cam-vreg-min-voltage = <2050000 1800000> ;
+		qcom,cam-vreg-max-voltage = <2050000 1800000>;
+		qcom,cam-vreg-op-mode = <2050000 1800000>;
 		pinctrl-names = "cam_default", "cam_suspend";
 		pinctrl-0 = <&cam_sensor_mclk0_default
 			     &cam_sensor_rear_default>;
 		pinctrl-1 = <&cam_sensor_mclk0_sleep &cam_sensor_rear_sleep>;
 		gpios = <&msm_gpio 26 0>,
-			<&msm_gpio 36 0>;
+			<&msm_gpio 36 0>,
+			<&msm_gpio 113 0>;
 		qcom,gpio-reset = <1>;
-		qcom,gpio-req-tbl-num = <0 1>;
-		qcom,gpio-req-tbl-flags = <1 0>;
+		qcom,gpio-vdig = <2>;
+		qcom,gpio-req-tbl-num = <0 1 2>;
+		qcom,gpio-req-tbl-flags = <1 0 0>;
 		qcom,gpio-req-tbl-label = "CAMIF_MCLK0",
-					  "CAM_RESET0";
+					  "CAM_RESET0",
+					  "CAM_VDIG0";
 		qcom,sensor-position = <0>;
 		qcom,sensor-mode = <0>;
 		qcom,cci-master = <0>;
@@ -67,7 +72,7 @@
 		clocks = <&clock_gcc clk_mclk0_clk_src>,
 			 <&clock_gcc clk_gcc_camss_mclk0_clk>;
 		clock-names = "cam_src_clk", "cam_clk";
-};
+	};
 
 	qcom,camera@1 {
 		cell-index = <1>;
-- 
2.11.0

